server/src/test/java/org/elasticsearch/index/mapper/IpPrefixAutomatonUtilTests.java

```diff
@@ -76,94 +77,105 @@ public void testIncompleteDecimalGroupAutomaton() throws UnknownHostException {
+            CompiledAutomaton a = buildIpPrefixAutomaton("10");
+            assertFalse(accepts(a, "1.2.3.4"));
+            assertTrue(accepts(a, "10.2.3.4"));
+            assertFalse(accepts(a, "2.2.3.4"));
+            assertFalse(accepts(a, "1::1"));
+            assertTrue(accepts(a, "10::1"));
+            assertTrue(accepts(a, "100::1"));
+            assertTrue(accepts(a, "1000::1"));
+            assertTrue(accepts(a, "1000::1.2.3.4"));
         }
         {
-            CompiledAutomaton compiledAutomaton = IpPrefixAutomatonUtil.buildIpPrefixAutomaton("1");
-            byte[] encode = InetAddressPoint.encode(InetAddress.getByName("1.2.3.4"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("10.2.3.4"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("2.2.3.4"));
-            assertFalse(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("1af::1:2"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("1f::1:2"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("::1:2"));
-            assertFalse(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("1cce:e003:0:0:9279:d8d3:ffff:ffff"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
+            CompiledAutomaton a = buildIpPrefixAutomaton("1");
+            assertTrue(accepts(a, "1.2.3.4"));
+            assertTrue(accepts(a, "10.2.3.4"));
+            assertFalse(accepts(a, "2.2.3.4"));
+            assertTrue(accepts(a, "1af::1:2"));
+            assertTrue(accepts(a, "1f::1:2"));
+            assertFalse(accepts(a, "::1:2"));
+            assertTrue(accepts(a, "1cce:e003:0:0:9279:d8d3:ffff:ffff"));
 
         }
         {
-            CompiledAutomaton compiledAutomaton = IpPrefixAutomatonUtil.buildIpPrefixAutomaton("1.");
-            byte[] encode = InetAddressPoint.encode(InetAddress.getByName("1.2.3.4"));
-            assertTrue(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("10.2.3.4"));
-            assertFalse(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
-            encode = InetAddressPoint.encode(InetAddress.getByName("2.2.3.4"));
-            assertFalse(compiledAutomaton.runAutomaton.run(encode, 0, encode.length));
+            CompiledAutomaton a = buildIpPrefixAutomaton("0");
+            assertTrue(accepts(a, "0.2.3.4"));
```

Comment:
We want ip4 to match "0" prefixes.

Reply:
Maybe have an explicit `assertFalse(accepts(a, "0000::127.0.0.1"))` here, given that's the specific case we are trying to avoid?